src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -205,24 +196,55 @@ public class DosFileManager {
             if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Error)) {
                 _loggerService.Error(e, "An error occurred while seeking file {@Error}", e);
             }
-            return DosFileOperationResult.Error(0x19);
+            return DosFileOperationResult.Error(ErrorCode.InvalidHandle);
         }
     }
 
     public DosFileOperationResult OpenFile(string fileName, byte rwAccessMode) {
+        string openMode = _fileOpenMode[rwAccessMode];
+
+        if (_dos.Devices.FirstOrDefault(virtualDevice => virtualDevice is CharacterDevice characterDevice && characterDevice.Name == fileName) is CharacterDevice device) {
```

Comment:
Can "is CharacterDevice device" twice be avoided in this line ?

Also until now LINQ was avoided for performance reasons (in emulation code, not reverse engineering APIs). But since .NET 6, LINQ so fast (and LINQ is so useful in general), some exceptions here or there are no big deal.